mbuild/compound.py

```diff
@@ -954,9 +954,31 @@ def bonds(self):
         else:
             return iter(())
 
+    @property
+    def direct_bonds(self):
+        """Returns the number of bonds a particle is directly involved in.
+        This method should only be used on on compounds at the bottom
+        of their hierarchy (i.e. a particle).
+
+        Returns
+        -------
+        int
+            The number of compounds this compound is directly bonded to.
+        """
+        if list(self.particles()) != [self]:
+            raise MBuildError(
+                "The direct_bonds method can only "
+                "be used on compounds at the bottom of their hierarchy."
+            )
+        if self.root.bond_graph.has_node(self):
```

Comment:
I think we also need to handle the case when the Compound has no bond, in which case, `self.root.bond_graph` will be `None` and `self.root.bond_graph.has_node(self)` will return an error. Adding one more elif clause should be enough to catch that (so I think 
```python3 
if not self.root.bond_graph:
     return 0
elif if self.root.bond_graph.has_node(self):
    ...
```

Reply:
we would also need to add a test case for when that check gets added.

Reply:
Ok, I think these scenarios are covered, and tested for.